### DISCUSSION TITLE
Tags Support

### DISCUSSION THREAD

**Discussion by Luctins** [gollum/gollum#1967] [Ideas]
Is there any way to display, or even better convert and display tags from the original markup?

I use the Org Mode adapter, and it would be very nice to be able to tag headings from the markup.

Something like:

```org
#+TITLE: Example

* Some Heading :some_tag:
Content
```

turns to

```html
<h1>Example</h1>

<h2>Some Heading (some_tag)</h2>
<p>Content</p>
```

My expertise is not web, so the style is just a suggestion, but macros for example could convert tags to emoji, to tag the language of someting for example: `:pt_br:` -> 🇧🇷 .

Thanks a lot for this great tool :).

**Comment by benjaminwil** [comment]
Hey there. I may be misunderstanding, but to my knowledge Gollum doesn't have any way to tie a tag to a specific article and allow users to browse articles by tag.

I can imagine someone adding specific markup to their pages, i.e. `:some_tag:` or `tag:some_tag` and then locating all articles with that tag using a conventional search. 

A custom macro could be used to automatically style these special tags. For more information about custom macros see [Custom Macros](https://github.com/gollum/gollum/wiki#custom-macros) in the Gollum wiki.

**Reply by Luctins** [reply]
Yeah, AFAIK article tagging was not a thing (would be very nice, but I know it's a lot of work). My goal was mainly to 'show through'  the tags from the markup on the headings. 

I'll take a look at the custom macros section and report back later, thanks for the tip.

**Reply by benjaminwil** [reply]
Just re-reading my comment and want to add to it to be more explicit.

You'll likely want a custom macro that converts the tag into some special HTML, like:

```
<span class="my-tag-class">
  some tag!
</span>
```

And use [a custom CSS stylesheet](https://github.com/gollum/gollum/wiki/Custom-Styling-and-Views) to provide styles for this HTML.
